dev/com.ibm.ws.concurrent.persistent_fat_multiple/fat/src/com/ibm/ws/concurrent/persistent/fat/multiple/MultiplePersistentExecutorsTest.java

```diff
@@ -0,0 +1,280 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.concurrent.persistent.fat.multiple;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeTrue;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.config.PersistentExecutor;
+import com.ibm.websphere.simplicity.config.ServerConfiguration;
+
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+
+/**
+ * Tests for multiple persistent scheduled executor instances sharing the same database
+ */
+@RunWith(FATRunner.class)
+public class MultiplePersistentExecutorsTest extends FATServletClient {
+
+    private static final String APP_NAME = "persistmultitest";
+    private static final Set<String> appNames = Collections.singleton(APP_NAME);
+    
+	@ClassRule
+    public static final JdbcDatabaseContainer<?> testContainer = DatabaseContainerFactory.create();
+	public static final DatabaseContainerType dbContainerType = DatabaseContainerFactory.usingContainerType();
+    
+    private static ServerConfiguration originalConfig;
+    
+    private static final LibertyServer server = FATSuite.server;
+    
+    private static final String TASK_ID_SEARCH_TEXT = "Task ids: ";
+    
+	@BeforeClass
+    public static void setUp() throws Exception {
+		//Skip if we do not have test container to run against
+		assumeTrue(dbContainerType.hasTestContainer());
```

Comment:
how does assumeTrue behave in a BeforeClass method? If the assumption is false, do all the tests skip? I would imagine we still want these tests to run even on Derby

Reply:
Yes, as soon as the assumption fails test execution ends.  The only thing that is run after that is cleanup code.  Derby is marked as having a test container since we have the noop container.  But I think I will remove the hasTestContainer attribute in favor of having the FAT fail if say "fat.bucket.db.type=Informix" or any other database that is not set up for the database rotation.